modulemd/modulemd-defaults.c

```diff
@@ -524,6 +524,60 @@ modulemd_defaults_new_from_string (const gchar *yaml_string, GError **error)
 }
 
 
+/**
+ * modulemd_defaults_new_from_stream:
+ * @yaml_file: A YAML stream containing the module metadata and other related
+ * information such as default streams.
+ * @error: (out): A #GError containing additional information if this function
+ * fails.
+ *
+ * Constructs a new #ModulemdDefaults object from the first valid
+ * modulemd-defaults document in the given module stream. This will ignore any
+ * documents of other types, malformed documents and defaults that appear later
+ * in the stream.
+ *
+ * Returns: A #ModulemdDefaults object constructed the first valid
+ * modulemd-defaults document in the given module stream. This must be freed
+ * with g_object_unref() when no longer needed.
+ *
+ * Since: 1.4
+ */
+ModulemdDefaults *
+modulemd_defaults_new_from_stream (FILE *stream, GError **error)
+{
+  GObject *object = NULL;
+  GPtrArray *data = NULL;
```

Comment:
`g_autoptr()`?

Reply:
Ah, I basically just copied this from the `modulemd_defaults_new_from_file()` routine. But you're right; I've been using `g_autoptr()` for all newly-written code; best not to break with that here. I will fix it up.